src/plugins/es_ui_shared/public/request/use_request.helpers.tsx

```diff
@@ -0,0 +1,181 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { mount, ReactWrapper } from 'enzyme';
+import sinon from 'sinon';
+
+import { HttpSetup, HttpFetchOptions } from '../../../../../src/core/public';
+import { SendRequestConfig, SendRequestResponse } from './send_request';
+import { useRequest, UseRequestResponse } from './use_request';
+
+export interface UseRequestHelpers {
+  advanceTime: (ms: number) => Promise<void>;
+  completeRequest: () => Promise<void>;
+  hookResult: UseRequestResponse;
+  getSendRequestSpy: () => sinon.SinonStub;
+  setupSuccessRequest: (overrides?: {}) => void;
+  getSuccessResponse: () => SendRequestResponse;
+  setupErrorRequest: (overrides?: {}) => void;
+  getErrorResponse: () => SendRequestResponse;
+  setErrorResponse: () => void;
+  setupErrorWithBodyRequest: (overrides?: {}) => void;
+  getErrorWithBodyResponse: () => SendRequestResponse;
+}
+
+// Each request will take 1s to resolve.
+export const REQUEST_TIME = 1000;
+
+const successRequest: SendRequestConfig = { method: 'post', path: '/success', body: {} };
+const successResponse = { statusCode: 200, data: { message: 'Success message' } };
+
+const errorValue = { statusCode: 400, statusText: 'Error message' };
+const errorRequest: SendRequestConfig = { method: 'post', path: '/error', body: {} };
+const errorResponse = { response: { data: errorValue } };
+
+const errorWithBodyRequest: SendRequestConfig = {
+  method: 'post',
+  path: '/errorWithBody',
+  body: {},
+};
+const errorWithBodyResponse = { body: errorValue };
+
+export const createUseRequestHelpers = (): UseRequestHelpers => {
+  // The behavior we're testing involves state changes over time, so we need finer control over
+  // timing.
+  jest.useFakeTimers();
+
+  const flushPromiseJobQueue = async () => {
+    // See https://stackoverflow.com/questions/52177631/jest-timer-and-promise-dont-work-well-settimeout-and-async-function
+    await Promise.resolve();
+  };
+
+  const completeRequest = async () => {
+    await act(async () => {
+      jest.runAllTimers();
+      await flushPromiseJobQueue();
+    });
+  };
+
+  const advanceTime = async (ms: number) => {
+    await act(async () => {
+      jest.advanceTimersByTime(ms);
+      await flushPromiseJobQueue();
+    });
+  };
+
+  let element: ReactWrapper;
+  // We'll use this object to observe the state of the hook and access its callback(s).
+  const hookResult = {} as UseRequestResponse;
+  const sendRequestSpy = sinon.stub();
+
+  const setupUseRequest = (config: SendRequestConfig) => {
+    const httpClient = {
```

Comment:
I originally moved it inside for clarity and readability, since the performance hit seems negligible. By placing it inside `setupUseRequest` it makes it clear that this is the only place it's consumed.